File: SymmetricMountains.py
# Get number of mountains
N = int(input())

# Get mountain heights
h = [int(i) for i in input().split()]

# Answer array
ans = []

# Populate array
for i in range(N):
    
    # Java's integer max value
    ans.append(2147483647)
    
# Iterate over mountains (for even crop values)
for i in range(N):
    
    # Store cumulative asymmetric value for this index
    crop_val = 0
    
    # Iterate over crop values (increment by 2)
    crop=0
    while i-crop//2>=0 and i+crop//2<N:
        
        # Calculate absolute difference, add to cumulative asymmetric value
        crop_val+=abs(h[i-crop//2]-h[i+crop//2])
        
        # Compare minimum asymmetric value for this crop with current cumulative asymmetric value
        ans[crop]=min(ans[crop], crop_val)
        
        # Increment crop by 2
        crop+=2
        
# Iterate over mountains (for odd crop values)
for i in range(N):
    
    # Store cumulative asymmetric value for this index
    crop_val = 0
    
    # Iterate over crop values (inceremnt by 2), this time starting at 1 (since odd crop values)
    crop=1
    while i-crop//2>=0 and i+crop//2+1<N:
        
        # Calculate absolute difference, add to cumulative asymmetric value
        crop_val+=abs(h[i-crop//2]-h[i+crop//2+1])
        
        # Compare minimum asymmetric value for this crop with current cumulative asymmetric value
        ans[crop]=min(ans[crop], crop_val)
        
        # Increment crop by 2
        crop+=2
        
# Output
for i in range(N):
    print(ans[i], end=" ")
        
    

File: SymmetricMountains.java
import java.util.*;

public class SymmetricMountains {
    
    public static void main(String[] args){

        // Create scanner
        Scanner input = new Scanner(System.in);

        // Number of mountains
        int N = input.nextInt();

        // Array of mountains
        int[] h = new int[N];

        // Get mountain heights
        for(int i=0; i<N; i++){
            h[i]=input.nextInt();
        }

        // Create array to store minimum asymmetric value for each crop
        int[] ans = new int[N];

        // Fill array with maximum integer values
        for(int i=0; i<N; i++){
            ans[i]=Integer.MAX_VALUE;
        }

        // Iterate over all mountains (to handle even crops)
        for(int i=0; i<N; i++){
            
            // Store cumulative asymmetric value at this index
            int crop_val = 0;

            // Iterate through even crop values
            for(int crop=0; i-crop/2>=0&&i+crop/2<N; crop+=2){

                // Calculate absolute difference, add to cumulative asymmetric value 
                crop_val+=Math.abs(h[i-crop/2]-h[i+crop/2]);

                // Compare minimum asymmetric value for this crop (thus far) to current 
                // cumulative asymmetric value
                ans[crop]=Math.min(ans[crop], crop_val);
            }
        }

        // Iterate over all mountains (to handle odd crops)
        for(int i=0; i<N; i++){

            // Store cumulative asymmetric value at this index
            int crop_val = 0;

            // Iterate through odd crop values
            for(int crop=1; i-crop/2>=0&&i+crop/2+1<N; crop+=2){

                // Calculate absolute difference, add to cumulative asymmetric value
                crop_val+=Math.abs(h[i-crop/2]-h[i+crop/2+1]);

                // Compare minimum asymmetric value for this crop (thus far) to current
                // Cumulative asymmetric value
                ans[crop]=Math.min(ans[crop], crop_val);
            }
        }

        // Output
        for(int i=0; i<N; i++){
            System.out.printf("%d ", ans[i]);
        }

    }
}
